### ISSUE TITLE
ZeroDivisionError in sum_circle when source is out hte image.

### ISSUE THREAD

**Issue by juliotux** [kbarbary/sep#62]
Hello,

I'm using a image simulated by Astromatic's SkyMaker to test a pipeline I'm developing. For some things, I use directly the sources list generated by SkyMaker as the input coordinates of sum_circle.

However, when I do this, sum_circle raises:
```
sep.pyx in sep.sum_circle (sep.c:9659)()

ZeroDivisionError: float division
```

This problem do not occur when I find the sources with the code. So, I think this problem may be generated by sources out of the image.

I a real situation, this problem appear when I have a set of images with bad tracking and try to use the coordinates of the first image (corrected by the shift) in the aperture photometry in all the dataset.

Any ideia on how this can be handled? Should not be better to the code, instead of raises an error, return nan values when it cannot compute the aperture photometry?

Thanks

**Comment by kbarbary** [comment]
Looking at the code, this happens when the background annulus has zero area (when using the `bkgann` option). This would indeed happen when the source is out of the image.

By default, cython makes division work like in pure Python, where `1.0 / 0.0` raises an error. I could change it to make it (1) emit a warning instead of an error, as in numpy (`np.array(1.) / np.array(0.)`) or (2) no warning or error.

I think maybe (1) would be better. People are used to this behavior from numpy.

 I think the return value would be `-inf` in either case. Returning `nan` would be nicer, but this would require more work - I'd want to ensure that apertures *not* using `bkgann` also return nan when entirely off the image. I don't think that happens now.

**Comment by juliotux** [comment]
When the source is out of the image and not use bkgann, the value returned is 0.0.

```
>>> import sep
>>> import numpy as np
>>> a = np.ones((64,64))
>>> flux, error, flags = sep.sum_circle(a, [300], [300], 5)
>>> print(flux)
[0.]
```

I think checking the `area1` variable (if it is zero) and put things inside ifs should be enough. Otherwise, checking if the source is inside the image before calculate is another option.

**Comment by juliotux** [comment]
Issue closed by #63
